Sprintfinity3902/Sprites/HudIcons/ItemSelectIconSprite.cs

```diff
@@ -23,9 +23,11 @@ public ItemSelectIconSprite(Texture2D texture)
             Texture = texture;
 
             Animation = new Animation();
+            /*TODO: This doesn't work and I don't know why*/
             Animation.AddFrame(sprite1, 0);
-            Animation.AddFrame(sprite2, 1/16f);
-            Animation.AddFrame(sprite1, 1/8f);
+            Animation.AddFrame(sprite2, 1/4f);
+            Animation.AddFrame(sprite1, 1/4f);
```

Comment:
Hey Ben, the problem is actually not here. Here are a few notes.

Firstly, you do need to stagger the frametimes, so it should read like it did before:
Animation.AddFrame(sprite1, 0);
Animation.AddFrame(sprite1, 1/8f);
Animation.AddFrame(sprite1, 1/4f);
If it doesn't, the animation will never play because frame 2 and frame 3 are essentially set to appear at the same frametime... which breaks Animation because it never finds the final SpriteFrame that it uses to loop around.

Secondly, the animation was not working because 
1. The Update() method within the InventoryHud is empty... so it never calls for the itemSelectedIcon to be updated. This means that the Animation will never be incremented, because this is done through a SpriteFrame's Update method. So, add something like itemSelectedIcon.Update(gameTime) to the update method. (Or, just add itemSelectedIcon and all the other icons to the list of IEntities in each Hud called Icons... that is why it is there, to house all of the icons, update and draw them)
2. Within Game1, when the GameState = Paused, the Hud's are not Updated. Similar to the PLAYING state, you need to add something to iterate through all the Huds and Update them: foreach(IHud hud in huds){ hud.Update(gameTime); }

Make these changes, and this Animation will play correctly.

Reply:
Oh also, you don't need to call Animation.Play(); at the end... I actually think this will set it to only play once. Delete that line.